container/state.go

```diff
@@ -324,13 +364,17 @@ func (s *State) SetRemovalInProgress() bool {
 // ResetRemovalInProgress makes the RemovalInProgress state to false.
 func (s *State) ResetRemovalInProgress() {
 	s.Lock()
+	oldState := s.StateString()
+	defer func() { s.counter.SetCount(oldState, s.StateString()) }()
 	s.RemovalInProgress = false
 	s.Unlock()
 }
 
 // SetDead sets the container state to "dead"
 func (s *State) SetDead() {
 	s.Lock()
+	oldState := s.StateString()
+	defer func() { s.counter.SetCount(oldState, s.StateString()) }()
 	s.Dead = true
 	s.Unlock()
```

Comment:
Isn't this still racy if the container changes state between `oldState := s.StateString()` and the defer running?

It would reduce the count of the old state twice?